src/hdr_histogram.c

```diff
@@ -666,29 +675,40 @@ int64_t hdr_min(const struct hdr_histogram* h)
     return non_zero_min(h);
 }
 
+static int64_t get_value_from_idx_up_to_count(const struct hdr_histogram* h, int64_t count_at_percentile ) {
+    int64_t count_to_idx = 0;
+    int64_t value_from_idx = 0;
+    int32_t sub_bucket_idx = -1;
+    int32_t bucket_idx = 0;
+    int32_t bucket_base_idx = get_bucket_base_index(h, bucket_idx);
+
+    for (;;) {
+        if (count_to_idx >= count_at_percentile) {
```

Comment:
I'm concerned about this being the only exit clause for the loop.  I think the sub_bucket_idx should also be checked to make sure that it doesn't run of the end or get into an infinite loop.

Reply:
Taken from here: https://github.com/HdrHistogram/hdrhistogram-go/blob/494271c4c016b36c8cee88480288f33b419cf7b0/hdr.go#L350-L354

@filipecosta90 Can you check if there's an issue here?

Reply:
@mikeb01 isn't the following check and change enough?:
https://github.com/HdrHistogram/HdrHistogram_c/pull/107#diff-3b9d3dbfd2aefc19e97618e6ca54cdab9ec1900faef90b1a450acad891980717R694
notice when we cross the sub-bucket limit we change bucket:
```c
        if (sub_bucket_idx >= h->sub_bucket_count)
        {
            sub_bucket_idx = h->sub_bucket_half_count;
            bucket_idx++;
            bucket_base_idx = get_bucket_base_index(h, bucket_idx);
        }
```

Reply:
@mikeb01 I think that since `count_at_percentile` is always less or equal to `total_count` then we can safely assume that `count_to_index` will reach or pass `count_at_percentile` before the call to `get_count_at_index_given_bucket_base_idx` is called with an overflowing `bucket_base_idx, sub_bucket_idx`. So I think it's safe.
@filipecosta90 please confirm.

Reply:
I added an overflow check that'll abort early if the passed `count_at_percentile` is to high. Please re-review.

Reply:
I think it works as long as all of the logic is correct and the total_count is consistent with the underlying data and nothing gets accidentally messed up.  "I think" being the operative term here.  My concern is that it takes a substantial mental effort and assumption of behaviour to verify correctness.  I would feel more confident with a simple check of the index against counts_len (the iterator move_next does this as an additional safety step).  I know this means a bit of restructuring of the code, e.g. in-lining `get_count_at_index_given_bucket_base_idx` so the index can be checked, but I think it will be safer and less fragile in the long run.